src/dwarf.rs

```diff
@@ -966,6 +1022,103 @@ impl DwarfResolver {
     }
 }
 
+struct SymbolInfo {
+    name: String,
+    address: u64,
+}
+
+fn parse_die_subprogram(
+    die: &mut debug_info::DIE<'_>,
+    str_data: &[u8],
+) -> Result<Option<SymbolInfo>, Error> {
+    let mut addr: Option<u64> = None;
+    let mut name_str: Option<&str> = None;
+    for (name, _form, _opt, value) in die {
+        match name {
+            constants::DW_AT_linkage_name | constants::DW_AT_name => {
+                if name_str.is_some() {
+                    continue;
+                }
+                match value {
+                    debug_info::AttrValue::Unsigned(str_off) => {
+                        match extract_string(&str_data, str_off as usize) {
+                            Some(s) => {
+                                name_str = Some(s);
+                            }
+                            _ => {
+                                return Err(Error::new(
+                                    ErrorKind::InvalidData,
+                                    "fail to extract the name of a subprogram",
+                                ));
+                            }
+                        }
+                    }
+                    debug_info::AttrValue::String(s) => {
+                        name_str = Some(s);
+                    }
+                    _ => {
+                        return Err(Error::new(
+                            ErrorKind::InvalidData,
+                            "fail to parse DW_AT_linkage_name {}",
+                        ));
+                    }
+                }
+            }
+            constants::DW_AT_lo_pc => match value {
+                debug_info::AttrValue::Unsigned(pc) => {
+                    addr = Some(pc);
+                }
+                _ => {
+                    return Err(Error::new(
+                        ErrorKind::InvalidData,
+                        "fail to parse DW_AT_lo_pc",
+                    ));
+                }
+            },
+            _ => {}
+        }
+    }
+
+    if addr.is_some() && name_str.is_some() {
+        Ok(Some(SymbolInfo {
+            name: name_str.unwrap().to_string(),
+            address: addr.unwrap(),
+        }))
+    } else {
+        Ok(None)
+    }
+}
+
+/// Parse the addresses of symbols from the `.debug_info` section.
+fn parse_symbols(parser: &Elf64Parser) -> Result<Vec<SymbolInfo>, Error> {
+    let info_sect_idx = parser.find_section(".debug_info")?;
+    let info_data = parser.read_section_raw(info_sect_idx)?;
+    let abbrev_sect_idx = parser.find_section(".debug_abbrev")?;
+    let abbrev_data = parser.read_section_raw(abbrev_sect_idx)?;
+    let units = debug_info::UnitIter::new(&info_data, &abbrev_data);
+    let str_sect_idx = parser.find_section(".debug_str")?;
+    let str_data = parser.read_section_raw(str_sect_idx)?;
+
+    let mut syms = Vec::<SymbolInfo>::new();
+    for (uhdr, dieitr) in units {
+        match uhdr {
+            debug_info::UnitHeader::CompileV4(_) => {
```

Comment:
what's CompileV4?

Reply:
DWARFv4 and DWARFv5 have Compile Unit Header in different format.  CompileV4 is the Compile Unit Header for DWARFv4.